### PR TITLE
Bump lib-common and services API to v0.1.0 for dev preview

### PR DESCRIPTION
**Pull request by fmount**
Bump lib-common modules to v0.1.0 tag and cleanup
go.mod file comments.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -236,22 +236,22 @@ github.com/openshift/api v0.0.0-20230414143018-3367bc7e6ac7 h1:rncLxJBpFGqBztyxC
 github.com/openshift/api v0.0.0-20230414143018-3367bc7e6ac7/go.mod h1:ctXNyWanKEjGj8sss1KjjHQ3ENKFm33FFnS5BKaIPh4=
 github.com/openstack-k8s-operators/cinder-operator/api v0.0.0-20230705113934-710f05ea0097 h1:3HIatVxA2SFDEkW5o3K1XlsKWdcln4vX0cvekN/P3+U=
```
Comment (fmount) [inline go.sum:237]: need to unblock this when cinder tag lands

Comment (konan-abhi) [reply go.sum:237]: ack!

**Review by konan-abhi — APPROVED** [review]
Looks good!

**Pull request merged**